SpringProject/src/main/java/com/gradproject2019/users/service/UserServiceImpl.java

```diff
@@ -118,7 +115,15 @@ private void checkIfManagerExists(Long managerId) {
 
     private void checkTokenMatchesUser(UUID token, Long userId) {
         if (!authServiceImpl.getTokenById(token).getUserId().equals(userId)) {
-            throw new UserUnauthorisedException();
+            throw new UserForbiddenException();
+        }
+    }
+
+    private void checkUserRequestingIsAuthorized(Long requestedUserId, UUID token) {
+        User requestingUser = getUserById(authServiceImpl.getTokenById(token).getUserId());
+        getUserById(requestedUserId);
```

Comment:
Is this still needed if the check is now done in the SQL statement?

Reply:
Well the second line checks the user you're requesting exists and throws a 404 if not whereas the if statement logic throws a 403 so yes I think?

Reply:
You're right, they throw different exceptions